### PR TITLE
Rust create backup

### PR DESCRIPTION
**Pull request by benma**
More unit tests to come after if the concept is approved.

### CODE REVIEW

File: src/rust/bitbox02-rust/src/hww/api/backup.rs [hunk 1]
```diff
@@ -68,3 +68,125 @@ pub async fn check(
...
+        };
+        if !confirm::confirm(&params).await {
+            return Err(Error::COMMANDER_ERR_USER_ABORT);
+        }
+        if bitbox02::memory::set_seed_birthdate(timestamp).is_err() {
+            return Err(Error::COMMANDER_ERR_MEMORY);
+        }
+        timestamp
+    } else if let Ok(backup::CheckData { birthdate, .. }) = backup::check() {
+        // If adding new backup after initialized, we do not know the seed bithdate.
```
Comment (NickeZ) [inline src/rust/bitbox02-rust/src/hww/api/backup.rs:122]: *birthdate

File: src/rust/bitbox02/src/lib.rs [hunk 2]
```diff
@@ -227,3 +230,19 @@ pub fn sdcard_inserted() -> bool {
     let data = crate::testing::DATA.0.borrow();
     data.sdcard_inserted.as_ref().unwrap()()
 }
+
+pub fn format_datetime(timestamp: u32, timezone_offset: i32, date_only: bool) -> String {
+    let mut out = [0u8; 100];
```
Comment (x1ddos) [inline src/rust/bitbox02/src/lib.rs:237]: I think 50 would've been more than enough?

**Review by x1ddos — APPROVED** [review]
Looks nice! Please continue :)

File: src/rust/bitbox02/src/lib.rs (hunk 2 above)
Comment (benma) [reply src/rust/bitbox02/src/lib.rs:237]: eh, i just copied what we had in C :P https://github.com/digitalbitbox/bitbox02-firmware/blob/d10f9f79e6c71b0da2e3ffab308e27fb3d862a10/src/workflow/confirm_time.c#L25

**Pull request merged**